dom.bs

```diff
@@ -4175,57 +4157,80 @@ returns as mask:
 
 <ol>
- <li>Let <var>reference</var> be the <a>context object</a>.
- <li>If <var>other</var> and <var>reference</var> are the
- same object, return zero.
+ <li><p>If <a>context object</a> is <var>other</var>, then return zero.
+
+ <li><p>Let <var>node1</var> be <var>other</var> and <var>node2</var> be <a>context object</a>.
+
+ <li><p>Let <var>attr1</var> be null.
 
  <li>
-  If <var>other</var> and <var>reference</var> are not
-  in the same <a>tree</a>, return the result of
-  adding
-  {{Node/DOCUMENT_POSITION_DISCONNECTED}},
-  {{Node/DOCUMENT_POSITION_IMPLEMENTATION_SPECIFIC}},
-  and either
-  {{Node/DOCUMENT_POSITION_PRECEDING}} or
-  {{Node/DOCUMENT_POSITION_FOLLOWING}},
-  with the constraint that this is to be consistent, together.
-
-  <p class="note no-backref">Whether to return
-  {{Node/DOCUMENT_POSITION_PRECEDING}} or
-  {{Node/DOCUMENT_POSITION_FOLLOWING}}
-  is typically implemented via pointer comparison. In JavaScript
-  implementations <code class='lang-javascript'>Math.random()</code> can be used.
-
- <li>If <var>other</var> is an
- <a>ancestor</a> of
- <var>reference</var>, return the result of adding
- {{Node/DOCUMENT_POSITION_CONTAINS}}
- to
- {{Node/DOCUMENT_POSITION_PRECEDING}}.
+  <p>If <var>node1</var> is an <a>attribute</a>, then:
+
+  <ol>
+   <li><p>Set <var>attr1</var> to <var>node1</var>.
 
- <li>If <var>other</var> is a
- <a>descendant</a> of
- <var>reference</var>, return the result of adding
- {{Node/DOCUMENT_POSITION_CONTAINED_BY}}
- to
- {{Node/DOCUMENT_POSITION_FOLLOWING}}.
+   <li><p>If <var>attr1</var>'s <a for=Attr>element</a> is non-null, then set <var>node1</var> to
+   <var>attr1</var>'s <a for=Attr>element</a>.
+  </ol>
 
- <li>If <var>other</var> is
- <a>preceding</a>
- <var>reference</var> return
+ <li>
+  <p>If <var>node2</var> is an <a>attribute</a>, then:
+
+  <ol>
+   <li><p>Let <var>attr2</var> be <var>node2</var>.
+
+   <li>
+    <p>If <var>attr2</var>'s <a for=Attr>element</a> is <var>node1</var> and <var>attr1</var> is
+    non-null, then:
+
+    <ol>
+     <li>
+      <p>For each <a>attribute</a> <var>node2Attribute</var> in <var>attr2</var>'s
+      <a for=Attr>element</a>'s <a for=Element>attribute list</a>:
+
+      <ol>
+       <li><p>If <var>node2Attribute</var> <a for=Node>equals</a> <var>attr1</var>, then return the
+       result of adding {{Node/DOCUMENT_POSITION_IMPLEMENTATION_SPECIFIC}} and
+       {{Node/DOCUMENT_POSITION_PRECEDING}}.
+
+       <li><p>If <var>node2Attribute</var> <a for=Node>equals</a> <var>attr2</var>, then return the
+       result of adding {{Node/DOCUMENT_POSITION_IMPLEMENTATION_SPECIFIC}} and
+       {{Node/DOCUMENT_POSITION_FOLLOWING}}.
+      </ol>
+    </ol>
+
+   <li><p>If <var>attr2</var>'s <a for=Attr>element</a> is non-null, then set <var>node2</var> to
+   <var>attr2</var>'s <a for=Attr>element</a>.
+  </ol>
+
+ <li>
+  <p>If <var>node1</var>'s <a for=tree>root</a> is not <var>node2</var>'s <a for=tree>root</a>, then
```

Comment:
This is a stellar API, I think the web platform should have more just like this.

On a more serious note, I think that when you're comparing two detached Attr nodes, you'll actually end up way down in "Return DOCUMENT_POSITION_FOLLOWING" instead of this ridiculous branch like implementations seem to.